### PR TITLE
Add Multiget Operations

### PR DESCRIPTION
**Pull request by jezell**

### CODE REVIEW

File: src/rocksdb.rs [hunk 1]
```diff
@@ -508,6 +518,117 @@ impl DB {
...
+                    let em = error_message(err);
+                    values.push(Err(em));
+                }
+
+                let val = value_ptrs[i];
+                if val.is_null() {
+                    values.push(Ok(None));
+                } else {
+                    values.push(Ok(Some(DBVector::from_c(val, length_ptrs[i]))));
+                }
```
Comment (spacejam) [inline src/rocksdb.rs]: I think it's important to maintain the 1:1 mapping between requested keys, and a single `Err`/`Ok` in `values`. Can we make the `if val.is_null() {` an `else if` after the check for errors, to guarantee a 1:1 mapping in returned values? (and the same for the multi_get_cf code below?)

File: src/rocksdb.rs [hunk 2]
```diff
@@ -975,6 +1096,36 @@ fn writebatch_works() {
...
+            _ =>  assert!(false)
+        }
+        match all[2] {
+            Ok(Some(ref u)) => assert_eq!(u.to_utf8(),Some("v3333")),
+            _ =>  assert!(false)
+        }
+    }
+    let opts = Options::default();
+    assert!(DB::destroy(&opts, path).is_ok());
+}
```
Comment (spacejam) [inline src/rocksdb.rs]: Can we add a test here that requests a key that is not present, and makes sure the 1:1 mapping between requested keys and the resulting `Result`?

File: src/rocksdb.rs (hunk 1 above)
Comment (spacejam) [inline src/rocksdb.rs]: could also continue the for loop in the error check if statement, or whatever you prefer to maintain 1:1 return values to requested keys.

File: src/rocksdb.rs (hunk 2 above)
Comment (jezell) [inline src/rocksdb.rs]: sure

File: src/rocksdb.rs (hunk 1 above)
Comment (jezell) [inline src/rocksdb.rs]: yeah, that's a good catch.

**Pull request closed without merging**